pkg/ebpf/tracee.go

```diff
@@ -580,6 +619,25 @@ func (t *Tracee) populateBPFMaps() error {
 		}
 	}
 
+	if t.kernelSymbols != nil {
+		// Initialize map for global symbols of the kernel
+		bpfKsymsMap, err := t.bpfModule.GetMap("ksymbols_map") // u32, u64
+		if err != nil {
+			return err
+		}
+
+		var reqKsyms []string
+		for id, event := range EventsDefinitions {
+			if event.EssentialEvent || t.eventsToTrace[id] {
+				reqKsyms = append(reqKsyms, event.Dependencies.ksymbols...)
+			}
+		}
```

Comment:
This looks like something you could have achieved in the EventsDefinitions iteration you did in New.
See my previous comment